### ISSUE TITLE
Getting `reading response header from upstream` error when trying to create new secrets with 1.7.2 on S3

### ISSUE THREAD

**Issue by jerrinss5** [PrivateBin/docker-nginx-fpm-alpine#192]
## Steps to reproduce
1. Install v1.7.2 - I tested it using docker container https://hub.docker.com/r/privatebin/nginx-fpm-alpine/ and tied to an S3 bucket
2. Create a new secret and it runs in to an error
3. Same doesn't happen with v.1.7.1

### What happens
```
2024/05/06 16:01:58 [error] 18#18: *7961 FastCGI sent in stderr: "PrivateBin\Controller->_create()
PrivateBin/PrivateBin#8 /var/www/index.php(18): PrivateBin\Controller->__construct()
PrivateBin/PrivateBin#9 {main}
  thrown in /srv/vendor/aws/aws-sdk-php/src/Credentials/CredentialProvider.php on line 72" while reading response header from upstream, client: <CUT>, server: , request: "POST / HTTP/1.1", upstream: "fastcgi://unix:/run/php-fpm.sock:", host: "<cut>"
```

### What should happen
Secret should be created without error

## Additional information
NA

## Basic information

**Server address**: BA

**Server OS: Docker container https://hub.docker.com/r/privatebin/nginx-fpm-alpine/**

**Webserver: Amazon Linux 2**

**Browser: NA**

**PrivateBin version: 1.7.2**

I can reproduce this issue on <https://privatebin.net>: No

**Comment by elrido** [comment]
Thank you for testing. So since this is an infra problem and not one with the PrivateBin software, I've transferred the issue over to the container image project. I currently see two solution paths:

- short term, I could simply add the missing module to the composer related packages in the 8.2 variant, so we can trick composer into still installing the later library - need to be careful not to break the edge build, though, where this is not necessary
- mid term, as in the next week(s), Alpine 3.20 will get released, at the level you currently get in edge, which avoids causing the issue and means the extra library can be removed again

longer term we need to take more care when switching php releases, and need to pay attention when composer is on a different release than the image targets. 8.4 (currently in development) will be our next opportunity to screw this up or deal with it more gracefully.

Maybe a multi-stage build could side-step the issue - download the release and composer artifacts in a build stage with git, wget, gnupg validation and composer, then transfer that application into the runtime container. Would also avoid the extra cleanup steps we currently have to do to remove composer & Co. from the image.

**Comment by jerrinss5** [comment]
Thank you @QuentinBtd for testing and @elrido for the analysis and fix suggestions

**Comment by elrido** [comment]
The fixed images are now available under tag `1.7.2-alpine3.19.1-s3-composer` and `1.7.2`, `stable` and `latest` got updated to point to it as well.

**Comment by jerrinss5** [comment]
Thanks @elrido - I can confirm that it works fine on the latest image.
